packages/log-view/src/browser/log-view-contribution.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { MenuContribution, CommandContribution, MenuModelRegistry, CommandRegistry, Command } from "@theia/core";
+import { WidgetManager, FrontendApplication, CommonMenus } from "@theia/core/lib/browser";
+import { inject, injectable, multiInject } from "inversify";
+import { LOG_VIEW_KIND } from "./log-view-widget";
+import { LanguageClientContribution } from "@theia/languages/lib/browser";
+import { LogModel } from "./log-model";
+import { LogMessageNotification } from "@theia/languages/lib/common";
+
+export namespace LogViewCommands {
+    export const OPEN: Command = {
+        id: 'logView:open',
+        label: 'Open Log View'
+    };
+}
+
+@injectable()
+export class LogViewContribution implements CommandContribution, MenuContribution {
+
+    constructor(
+        @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
+        @inject(FrontendApplication) protected readonly app: FrontendApplication,
+        @multiInject(LanguageClientContribution) languageClients: LanguageClientContribution[],
+        @inject(LogModel) logModel: LogModel) {
+        languageClients.forEach(contrib => {
+            contrib.languageClient.then(
```

Comment:
We should not do it like that but implement [Window.createOutputChannel](https://github.com/TypeFox/vscode-languageserver-node/blob/e0c1f194dabede0409f0d8f4e4e2eeccc65e1535/client/src/services.ts#L177). `LanguageClient` already is using it [to log messages posted by LogMessageNotification](https://github.com/TypeFox/vscode-languageserver-node/blob/e0c1f194dabede0409f0d8f4e4e2eeccc65e1535/client/src/base.ts#L910-L925) as well as additional messages about communication with particular language server.